lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -192,6 +192,27 @@ def retrying_submit(actions)
       end
     end
 
+    # we shouldn't send unsupported actions to ES
+    # method filters out unsupported actions by warning, writes event to DQL if enabled
+    # @returns valid actions
+    def filter_unsupported_actions(actions)
+      return if actions.nil? || actions.size < 1
+      supported_actions, unsupported_actions = actions.partition { |action, _, _| LogStash::Outputs::ElasticSearch::VALID_HTTP_ACTIONS.include?(action) }
+      if !unsupported_actions.nil? && unsupported_actions.size > 0
+        handle_dlq_or_drop(unsupported_actions)
+        @logger.warn("Number of requests filtered out before sending to Elasticsearch because of unsupported action, ", size: actions.size)
+      end
+      supported_actions
+    end
+
+    def handle_dlq_or_drop(unsupported_actions)
+      return if @dlq_writer.nil?
+
+      unsupported_actions.each do |action|
+        @dlq_writer.write(action.event, "Unsupported event.")
+      end
+    end
```

Comment:
Method-naming aside, since:
 - we _should_ log each and every instance of data-loss,
 - we _should_ increment the `non_retryable_failures` metric, AND
 - we will _always_ get an array,

this method becomes something like:

```suggestion
    def handle_dlq_or_drop(unsupported_actions)
      unsupported_actions.each do |action|
        if @dlq_writer
          @dlq_writer.write(action.event, "Unsupported action.")
        else
          @document_level_metrics.increment(:non_retryable_failures)
          @logger.warn("Could not index event to Elasticsearch because its action is not supported.", action: action)
        end
      end
    end
```

Reply:
`handle_dlq_status` method is now used as a common to handle such cases.